crypto/fipsmodule/cpucap/cpu_aarch64_apple.c

```diff
@@ -48,6 +48,25 @@ static int has_hw_feature(const char *name) {
   return value != 0;
 }
 
+// This function compares the brand retrieved with the input string
+// up to the length of the shortest of these 2 strings.
+static int is_brand(const char *in_str) {
+  char brand[64];
+  size_t len = sizeof(brand);
+  if (sysctlbyname("machdep.cpu.brand_string", brand, &len, NULL, 0) != 0 ||
+      strncmp(brand, in_str, strnlen(in_str, len)) != 0) {
+    return 0;
+  }
+
+  if (len > sizeof(brand)) {
```

Comment:
why check this here? The real problem would be if this condition is true when doing `strncmp()`.

See https://developer.apple.com/documentation/kernel/1387446-sysctlbyname
> If the amount of data is greater than the value in this parameter, the function updates this parameter to the required size and returns ENOMEM.